### PR TITLE
POS Vertically centre payment amounts between cash button and card reader view

### PR DESCRIPTION
**Pull request by joshheald**
## Description

This PR updates the design of the payment screen to match latest designs.

Specifically, we centre the totals between the cash button and the card payment prompts.

## Test Steps

Check both payment flows:

- Purchasing products
- Paying for bookings

Check that the amounts are vertically centred for various pre-payment scenarios:

- No reader attached
- Reader attached
- $0 orders

## Screenshots

https://github.com/user-attachments/assets/a3b9bc8c-ed2d-4fad-8799-b8e7c6270cc8

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/269">24.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by joshheald** [comment]
@iamgabrielma  @staskus It's not a big deal if this one gets bumped to the next release, if it doesn't get reviewed in time today.

**Comment by staskus** [comment]
@joshheald I started to test this but I know from experience how many combinations this one has, and this view had these corner cases where something wouldn't fit right. I'm currently generating all of these combinations for previews to make sure most of the cases are cover. However, I also want to start the release process, so this will be switched to the next version.

**Review by staskus — APPROVED** [review]
✅  Thanks. Works well. Tested various success and error cases, including large dynamic types.

I think the only thing that could be improved is the "No card reader connected" state. It's the only case where Cash Button doesn't appear at the bottom. 

Image

**Comment by joshheald** [comment]
> ✅ Thanks. Works well. Tested various success and error cases, including large dynamic types.
> 
> I think the only thing that could be improved is the "No card reader connected" state. It's the only case where Cash Button doesn't appear at the bottom.


Ooh, that's super weird. I looked at that screen a lot and saw nothing like that. I'll figure out what's happened and fix before merging – I wouldn't really expect there to be iOS 18/26 differences here!

**Comment by joshheald** [comment]
@staskus Could you check this disconnected reader issue again, when you get a moment, please? 

The only way I could repro this is on an iOS 26 sim, by changing the size of the window slowly until the layout breaks. However... when it does, the whole app breaks and becomes unresponsive, just a picture that gets stretched and squashed. I'm not 100% sure if it's a real repro (but... I've finally got an iOS 26 iPad arriving later this week!)

It seems like this comes from `scrollVerticallyIfNeeded` and some strangeness from iOS's layout engine during resizing, making it put the totals view in a scroll view unnecessarily, and collapsing the spacers in the process.

One way to fix this was to set the minimum height of the geometryListener to match its parent, but actually we don't need most of this modifier any more – it was for pre-iOS 17 behaviour when we couldn't remove the bounce behaviour if we had a scroll view when the content did fit.

So, I've kept the modifier but simplified it to a basic wrapper that always puts the scrollview in, but without so many of the old downsides. We still need a geometryReader to ensure that spacers don't collapse, but at least we've no more branching or comparing content/view heights as they change.


I'm going to test it a lot more tomorrow, but let me know what you think – change in 90daafa

File: Modules/Sources/WooFoundation/ViewModifiers/View+ScrollModifiers.swift [hunk 1]
```diff
@@ -1,53 +1,18 @@
...
-                /// Using `onPreferenceChange` avoid changing state (`shouldScroll`) during a layout pass.
-                /// Changing state that's used to layout the view during layout can cause an infinite loop and make the screen unresponsive.
-                if let contentSize = contentSize {
-                    shouldScroll = contentSize.height > parentGeometry.size.height
-                }
+            ScrollView(.vertical, showsIndicators: false) {
+                content
+                    .frame(minHeight: parentGeometry.size.height)
             }
+            .scrollBounceBehavior(.basedOnSize)
```
Comment (staskus) [inline Modules/Sources/WooFoundation/ViewModifiers/View+ScrollModifiers.swift:13]: 👍 nice, much better than a custom calculation logic... especially given it creates these sorts of issues. great finding

**Review by staskus — APPROVED** [review]
✅ Works well, the cash payment button appears correctly. 

Thank you for a change that simplifies and solves some glitches. Always good to avoid manual calculations. Tested the views that depended on 'scrollVerticallyIfNeeded`. 

Noticed it is missing in `PointOfSaleCardPresentPaymentScanningForReadersFailedView`, which could be added as well, although scroll only kicks in on iPad Mini with extremely large dynamic type, so it's a super nit.

**Comment by joshheald** [comment]
Added it there as well.

I've checked on the phone, in some of the shipping and Jetpack screens etc, and it's good to go.
